src/script/lua_api/l_nodemeta.cpp

```diff
@@ -254,42 +254,51 @@ int NodeMetaRef::l_from_table(lua_State *L)
 	// clear old metadata first
 	ref->m_env->getMap().removeNodeMetadata(ref->m_p);
 
-	if(lua_isnil(L, base)){
+	if (!lua_istable(L, base)) {
 		// No metadata
 		lua_pushboolean(L, true);
 		return 1;
 	}
 
 	// Create new metadata
 	NodeMetadata *meta = getmeta(ref, true);
-	if(meta == NULL){
+	if (meta == NULL) {
 		lua_pushboolean(L, false);
 		return 1;
 	}
+
 	// Set fields
 	lua_getfield(L, base, "fields");
-	int fieldstable = lua_gettop(L);
-	lua_pushnil(L);
-	while(lua_next(L, fieldstable) != 0){
-		// key at index -2 and value at index -1
-		std::string name = lua_tostring(L, -2);
-		size_t cl;
-		const char *cs = lua_tolstring(L, -1, &cl);
-		std::string value(cs, cl);
-		meta->setString(name, value);
-		lua_pop(L, 1); // removes value, keeps key for next iteration
+	if (lua_istable(L, -1)) {
```

Comment:
Note to reviewers: Only this line and the `lua_pop` call were added in both parts.